atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -0,0 +1,159 @@
+"""
+Copyright (c) 2020 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals, absolute_import
+
+import json
+import os
+
+from atomic_reactor.constants import (PLUGIN_ADD_IMAGE_CONTENT_MANIFEST,
+                                      REPO_CONTENT_SETS_CONFIG,
+                                      IMAGE_BUILD_INFO_DIR, INSPECT_ROOTFS,
+                                      INSPECT_ROOTFS_LAYERS)
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.util import (df_parser, read_yaml_from_file_path, base_image_is_scratch,
+                                 read_yaml)
+from osbs.utils import Labels
+
+
+class AddImageContentManifestPlugin(PreBuildPlugin):
+    """
+    Add the ICM JSON file to the IMAGE_BUILD_INFO_DIR/content_manifests
+    directory, for the current platform. Filename will be '{IMAGE_NVR}.json'
+
+    ICM examples:
+
+    WITHOUT content_sets specified:
+
+    {
+      "metadata": {
+        "icm_version": 1,
+        "icm_spec": "https://link.to.icm.specification",
+        "image_layer_index": 3
+      },
+      "content_sets" : [],
+      "image_contents": [
+        {
+          "purl": "pkg:golang/github.com%2Frelease-engineering%2Fretrodep%2Fv2@v2.0.2",
+          "dependencies": [{"purl": "pkg:golang/github.com%2Fop%2Fgo-logging@v0.0.0"}],
+          "sources": [{"purl": "pkg:golang/github.com%2FMasterminds%2Fsemver@v1.4.2"}]
+        }
+      ]
+    }
+
+    WITH content_sets specified:
+
+    {
+      "metadata": {
+        "icm_version": 1,
+        "icm_spec": "https://link.to.icm.specification",
+        "image_layer_index": 2
+      },
+      "content_sets": [
+          "rhel-8-for-x86_64-baseos-rpms",
+          "rhel-8-for-x86_64-appstream-rpms"
+      ],
+      "image_contents": [
+        {
+          "purl": "pkg:golang/github.com%2Frelease-engineering%2Fretrodep%2Fv2@v2.0.2",
+          "dependencies": [{"purl": "pkg:golang/github.com%2Fop%2Fgo-logging@v0.0.0"}],
+          "sources": [{"purl": "pkg:golang/github.com%2FMasterminds%2Fsemver@v1.4.2"}]
+        }
+      ]
+    }
+    """
+    key = PLUGIN_ADD_IMAGE_CONTENT_MANIFEST
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, remote_source_content_manifest,
```

Comment:
`remote_source_content_manifest` could be this too big to be transferred to workers via buildconfig? What is OCP limit?

Reply:
@MartinBasti; This is the design I worked from... @athos-ribeiro, @rcerven?

Reply:
In theory, the file could actualy be quite large. Thanks for spotting this, @MartinBasti.

@ben-alkov, we should use the same approach as the one used in https://github.com/containerbuildsystem/atomic-reactor/pull/1495 and defer downloading the ICM here in the worker; i.e., https://github.com/containerbuildsystem/atomic-reactor/pull/1503 should be changed to forward the ICM URL instead.